### ISSUE TITLE
Stop opening Staged Changes when selecting an open editor for a file that has already been staged

### ISSUE THREAD

**Issue by gregveres** [microsoft/vscode#207291]
I have run into a very annoying behavour of VS Code's source control panel. 
When I am working through a large change with many files, I will often look at a source file, do the review of that source file, make sure that the unit tests have been updated etc. 
Once I have dealt with the file and it's corresponding unit test file and related files, I will stage those files.
I keep the staged changes section closed so that I can focus on what is left in the "Changes" section. 
What I find extremely annoying is that the staged changes will continually "pop open". I just figured out that it is popping open because I inadvertently select an open editor that has a file that is staged. 

This is extremely disruptive to my workflow and never something I want to have happen. If I close the Staged Changes, I want it to stay closed so that I can focus on the unstaged changes. 

So my feature request is to stop opening the Staged Changes section when an editor with a staged file is selected.

**Comment by gregveres** [comment]
@lszomoru Yes, that stopped the behaviour. Thank you!

Now that I know there is an "auto reveal" feature, I would have to say that it is broken in two ways.

1. It shouldn't expand any closed tree items that it doesn't need to expand (this was the behaviour that caused me to file this issue report)
2. It needs to scroll the selected file into view. I had no idea that an auto reveal feature existed because my most of the time when I had selected a file, that file wasn't visible in the git viewport. The git viewport needed to scroll to get the file into view. 

But I really don't care if those two issues get fixed because I have turned off auto reveal and I don't expect to turn it back on.  I guess I would if those two were fixed, but I don't feel that I am missing any functionality by not having it enabled.
